Change setBody to be less strict when getting the immediate child

tokarzkj opened the pull request:
Step four of the dom document body calls for throwing an exception
if no document element was found. The current setBody implementation
was looking for a document element that is also an html element. It
has been updated to use GetDocumentElement which returns any document
element.

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #24291 

<!-- Either: -->
- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/24316)
<!-- Reviewable:end -->

tokarzkj committed:
Change setBody to be less strict when getting the immediate child

Step four of the dom document body calls for throwing an exception
if no document element was found. The current setBody implementation
was looking for a document element that is also an html element. It
has been updated to use GetDocumentElement which returns any document
element.

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @simonsapin (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @asajeffrey: components/script/dom/document.rs
 * @kichjang: components/script/dom/document.rs

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify script code, but no tests are modified. Please consider adding a test!

CYBAI commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit b567cc76e5adf00e24f46aa8b3337c3c3901ba46 with merge 9ceb3f6141ec3d3024894dc60d16e3e4f401ba76...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/13533)

CYBAI reviewed (approved):
Thanks for working on this! 🕺

The implementation looks right to me! and it did fix the test case as @jdm referred in the issue.

The last thing to do is to mark the test as `PASS`.

```
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "Setting document.body to a new body element when the root element is a test element.", 
    "test": "/html/dom/documents/dom-tree-accessors/Document.body.html", 
    "line": 300576, 
    "action": "test_result", 
    "expected": "FAIL"
}
```

After the test is `PASS`-ed, you can remove the FAIL test case in its metadata ini file. But, in this case, the test was the last FAIL test so you can just remove the `tests/wpt/metadata/html/dom/documents/dom-tree-accessors/Document.body.html.ini` file!

tokarzkj committed:
Remove the Document.body.html.ini file

This file was setting the test as expected to fail. However, after
the changes made in the pull request the test is now expected to
pass

jdm commented:
@bors-servo r+
Thank you for fixing this!

bors-servo commented:
:pushpin: Commit 6949052 has been approved by `jdm`

<!-- @bors-servo r=jdm 6949052094e774bd56e39857f087ec531b241404 -->

bors-servo commented:
:hourglass: Testing commit 6949052094e774bd56e39857f087ec531b241404 with merge 45b68ecbfc08738fd731383f6369725eef598105...

bors-servo commented:
:sunny: Test successful - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/13540), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/13602), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/EjlyB2hvSBWdySuRdiQHmw)
Approved by: jdm
Pushing 45b68ecbfc08738fd731383f6369725eef598105 to master...
